openapi.yml

```diff
@@ -0,0 +1,105 @@
+openapi: 3.0.3
+info:
+  version: 0.0.1
+  title: Order Book API
+servers:
+- url: http://localhost:8080
+  description: Local
+paths:
+  /api/v1/order:
+    post:
+      summary: Create a new order.
+      responses:
+        201:
+          description: Order has been accepted.
+      requestBody:
+        description: The order to create.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Order"
+  /api/v1/orders:
+    get:
+      summary: Get existing orders.
+      # TODO: Later we could add filters through the url query to only get orders matching specific tokens etc.
```

Comment:
Where should the "validation logic" (e.g. checking sufficient approval etc) live? Would we allow to filter this as well?

Reply:
I was thinking that the orderbook is going to have some automatic cleanup of stale orders. For example, it can always remove (and reject) orders with a nonce or timestamp in the past so there is no need for a filter there.

For the approval balance it is more difficult because they can change quickly and can start out as invalid but become valid. That looks like something better suited for the solver but not sure yet.